src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -76,7 +78,7 @@ object DeployPlugin extends AutoPlugin {
       */
     val deployDirs = settingKey[Seq[String]](
       "subdirectories from the stage task to copy during deploy. " +
-        "defaults to bin/, conf/, lib/, and public/"
```

Comment:
Rather than copying the whole of `conf/`, this now loads the appropriate environment configuration (i.e. `dev.conf`) in a way that fully resolves imports and substitutions, applies overrides to that `Config` object, then writes it alone to the replica's directory as `conf/env.conf`.